lms/templates/discussion/index.html

```diff
@@ -22,6 +22,13 @@
 ## Enable fast preview to fix discussion MathJax rendering bug when page first loads.
 <%include file="_js_body_dependencies.html" args="disable_fast_preview=False"/>
 <%static:js group='discussion'/>
+<script language="javascript">
+   $(function () {
+       $("section.discussion").each(function (index, elem) {
+           DiscussionApp.start(elem, true);
+       });
+   });
+ </script>
```

Comment:
@bjacobel "the 3rd party tool that should not be named" is a 3rd party tool some solutions client uses instead of LMS. And most of the fixes/changes I'm "forward-porting" in this PR are made to fix bugs caused or exposed by the use of that 3rd party tool.

I've been lectured a couple of times in the past to not mention clients of edX in public places like github and Slack, so I'm not naming it. If you're curious, ask Steve Magoun - he knows for sure who I am talking about.

Regarding this partiicular change - the only actual reason to bring it here is to reduce code drift. But it is relatively simple, so I believe such code drift is affordable. It looks like the best course of actions would be to remove this commit from this PR - I'll do that now and if you have any better ideas - let me know.